File: AIMario.CPlusPlus/src/hud.cpp
#include "hud.h"
#include "level.h"
#include "sprites.h"
#include <cstdio>

void DrawHUD(const SpriteGenerator& sprites, const Level& level) {
    Color hudWhite = {252,252,252,255};
    Color hudGold = {248,184,0,255};
    float y = 8.0f;

    Level::DrawTextColored(sprites, "MARIO", 24, y, 1, hudWhite);
    char buf[32];
    snprintf(buf, sizeof(buf), "%06d", level.score);
    Level::DrawTextColored(sprites, buf, 24, y+10, 1, hudWhite);

    snprintf(buf, sizeof(buf), "x%02d", level.coins);
    Level::DrawTextColored(sprites, buf, 96, y+10, 1, hudGold);

    Level::DrawTextColored(sprites, "WORLD", 144, y, 1, hudWhite);
    Level::DrawTextColored(sprites, "1-1", 152, y+10, 1, hudWhite);

    Level::DrawTextColored(sprites, "TIME", 200, y, 1, hudWhite);
    snprintf(buf, sizeof(buf), "%03d", (int)level.timer);
    Level::DrawTextColored(sprites, buf, 208, y+10, 1, hudWhite);
}

File: AIMario.Rust/src/hud.rs
use macroquad::prelude::*;
use crate::level::Level;
use crate::sprites::SpriteGenerator;

pub fn draw_hud(sprites: &SpriteGenerator, level: &Level) {
    let hud_white = Color::from_rgba(252, 252, 252, 255);
    let hud_gold = Color::from_rgba(248, 184, 0, 255);

    let y = 8.0;
    let scale = 1.0;

    // MARIO
    draw_text_colored(sprites, "MARIO", 24.0, y, scale, hud_white);
    draw_text_colored(sprites, &format!("{:06}", level.score), 24.0, y + 10.0, scale, hud_white);

    // Coins
    draw_text_colored(sprites, &format!("x{:02}", level.coins), 96.0, y + 10.0, scale, hud_gold);

    // WORLD
    draw_text_colored(sprites, "WORLD", 144.0, y, scale, hud_white);
    draw_text_colored(sprites, "1-1", 152.0, y + 10.0, scale, hud_white);

    // TIME
    draw_text_colored(sprites, "TIME", 200.0, y, scale, hud_white);
    draw_text_colored(sprites, &format!("{:03}", level.timer as i32), 208.0, y + 10.0, scale, hud_white);
}

fn draw_text_colored(sprites: &SpriteGenerator, text: &str, x: f32, y: f32, scale: f32, color: Color) {
    Level::draw_text_colored(sprites, text, x, y, scale, color);
}
